File: javascript/Ordenaçao.js
var input = require('fs').readFileSync('javascript/stdin', 'utf8');
//ESTUDAR
var sep = [' ', '\n'];
var line = input.split(new RegExp(sep.join('|'), 'g'));


    function particao(a, baixo, alto) {
        let pivo = a[alto];
        let i = (baixo - 1);
        for (let j = baixo; j < alto; j++) {
            if (a[j] < pivo) {
                i++;
                let temp = a[i];
                a[i] = a[j];
                a[j] = temp;
            }
        }
        let temp = a[i + 1];
        a[i + 1] = a[alto];
        a[alto] = temp;

        return i + 1;
    }

    function quicksort(a, baixo, alto) {
        if (baixo < alto) {
            let parte = particao(a, baixo, alto);
            quicksort(a, baixo, parte - 1);
            quicksort(a, parte + 1, alto);
        }
    }

    function ordenacao() {
        let a;
        let i;
        console.log("Entrada do tamanho da lista:");
        let tamanho = parseInt(line.shift());
        console.log(tamanho);
        a = new Array();
        console.log();
        

        console.log("Entrada dos elementos da lista: ");
        for (i = 0; i < tamanho; i++) {
            a[i] = parseInt(line.shift());
            //console.log(a[i]);
        }
        console.log(a + " ");
        console.log();

        quicksort(a, 0, tamanho - 1);

        console.log("A ordenaçao dos elementos eh: ");
        for (i = 0; i < tamanho; i++) {
            //console.log(a[i] + " ");
        }
        console.log(a + " ");
    }

ordenacao();

File: java/src/Ordenaçao.java
import java.util.Scanner;

public class Ordenaçao {

    int particao(int a[], int baixo, int alto) {
        int pivo = a[alto];
        int i = (baixo - 1);
        for (int j = baixo; j < alto; j++) {
            if (a[j] < pivo) {
                i++;
                int temp = a[i];
                a[i] = a[j];
                a[j] = temp;
            }
        }
        int temp = a[i + 1];
        a[i + 1] = a[alto];
        a[alto] = temp;

        return i + 1;
    }

    void quicksort(int a[], int baixo, int alto) {
        if (baixo < alto) {
            int parte = particao(a, baixo, alto);
            quicksort(a, baixo, parte - 1);
            quicksort(a, parte + 1, alto);
        }
    }

    public void ordenacao() {
        Scanner input = new Scanner(System.in);
        int a[], i;
        System.out.println("Entrada do tamanho da lista:");
        int tamanho = input.nextInt();
        a = new int[tamanho];

        System.out.println("Entrada dos elementos da lista: ");
        for (i = 0; i < tamanho; i++) {
            a[i] = input.nextInt();
        }
        Ordenaçao obj = new Ordenaçao();
        obj.quicksort(a, 0, tamanho - 1);

        System.out.println("A ordenaçao dos elementos eh: ");
        for (i = 0; i < tamanho; i++) {
            System.out.print(a[i] + " ");
        }

    }
}